NetCoreRestApi/DataLayer/Converters/ProductModelConverter.cs

```diff
@@ -1,27 +1,23 @@
-﻿using DataLayer.Models;
-using Common.Converter;
+﻿using Common.Converter;
+using Common.Converters;
 using DataLayer.Entities;
-using System.Collections.Generic;
+using DataLayer.Models;
+using System;
+using System.Linq.Expressions;
 
 namespace DataLayer.Converters
 {
-    public class ProductModelConverter : IConverter<ProductEntity, ProductModel>
+    public class ProductModelConverter<TId> : BaseConverter<ProductEntity<TId>, ProductModel<TId>>
     {
-        private readonly IConverter<CategoryEntity, CategoryModel> _categoryConverter;
-
-        public ProductModelConverter(IConverter<CategoryEntity, CategoryModel> categoryConverter)
+        private readonly IConverter<CategoryEntity<TId>, CategoryModel<TId>> _categoryConverter;
+        
+        public ProductModelConverter(IConverter<CategoryEntity<TId>, CategoryModel<TId>> categoryConverter)
```

Comment:
Remove TId from ProductModelConverter, ProductEntity and ProductModel. Tid should be in generic BaseEntity, BaseModel classes